scipy/stats/_continuous_distns.py

```diff
@@ -9033,6 +9034,101 @@ def _munp(self, n, b):
 
 rice = rice_gen(a=0.0, name="rice")
 
+class irwinhall_gen(rv_continuous):
+    r"""An Irwin-Hall (Uniform Sum) continuous random variable.
+
+    An `Irwin-Hall <https://en.wikipedia.org/wiki/Irwin-Hall_distribution/>`_ 
+    [1]_ [2]_ continuous random variable is the sum 
+    of :math`n` independent standard uniform r.v.'s. 
+    
+    %(before_notes)s
+
+    Notes
+    -----
+    Applications include `Rao's Spacing Test <https://jammalam.faculty.pstat.ucsb.edu/html/favorite/test.htm>`_, 
+    a more powerful alternative to the Rayleigh test 
+    when the data are not unimodal, and radar [3]_.
+
+    Conveniently, the pdf and cdf are the :math:`n`-fold convolution of 
+    the ones for the standard uniform distribution,
+    which are well known to be just cardinal B-splines of degree :math:`n-1` 
+    with knots evenly spaced from :math:`1` to :math:`n`.
+
+    The Bates distribution is simply the Irwin-Hall distribution scaled 
+    by :math:`1/n`: ``bates = irwinhall(n, name="bates", scale=1/n)``
+
+    
+    %(after_notes)s
+
+    References
+    ----------
+    .. [1] HALL PHILIP, "THE DISTRIBUTION OF MEANS FOR SAMPLES OF SIZE N DRAWN 
+            FROM A POPULATION IN WHICH THE VARIATE TAKES VALUES BETWEEN 0 AND 1, 
+            ALL SUCH VALUES BEING EQUALLY PROBABLE", 
+            Biometrika, Volume 19, Issue 3-4, December 1927, Pages 240-244, 
+            https://doi.org/10.1093/biomet/19.3-4.240
+    .. [2] J. O. IRWIN, "ON THE FREQUENCY DISTRIBUTION OF THE MEANS OF SAMPLES 
+            FROM A POPULATION HAVING ANY LAW OF FREQUENCY WITH FINITE MOMENTS, 
+            WITH SPECIAL REFERENCE TO PEARSON'S TYPE II", 
+            Biometrika, Volume 19, Issue 3-4, December 1927, Pages 225-239, 
+            https://doi.org/10.1093/biomet/19.3-4.225
+    .. [3] K. Buchanan, T. Adeyemi, C. Flores-Molina, S. Wheeland and D. Overturf, 
+            "Sidelobe behavior and bandwidth characteristics 
+            of distributed antenna arrays," 
+            2018 United States National Committee of 
+            URSI National Radio Science Meeting (USNC-URSI NRSM), 
+            Boulder, CO, USA, 2018, pp. 1-2.
+            https://www.usnc-ursi-archive.org/nrsm/2018/papers/B15-9.pdf
+
+    %(example)s
+    """
+
+    def _argcheck(self, n):
+        return (n > 0) & _isintegral(n) & np.isrealobj(n)
+    
+    def _get_support(self, n):
+        return 0, n 
+    
+    def _shape_info(self):
+        return [_ShapeInfo("n", True, (1, np.inf), (True, False))]
+    
+    def _munp(self, n, t):
+        return (sc.expm1(t)/t)**n
```

Comment:
OK, I verified that the MGF of the uniform distribution on the unit interval is $\frac{e^t - 1}{t}$ and that $M_{X+Y} = M_x \cdot M_y$, so $M_{nX} = M_x^n$. 

However, `_munp` is not the moment generating function; it is the raw moment about the origin. The first argument, `n`, is the order of the moment, not the shape parameter `n` of the distribution. Is that what you were expecting?

In any case, this fails the generic moment test:

```
_________________________________ test_moments[irwinhall-arg50-True-True-True-False] __________________________________

distname = 'irwinhall', arg = (10,), normalization_ok = True, higher_ok = True, moment_ok = True, is_xfailing = False

    @pytest.mark.slow
    @pytest.mark.parametrize('distname,arg,normalization_ok,higher_ok,moment_ok,'
                             'is_xfailing',
                             cases_test_moments())
    def test_moments(distname, arg, normalization_ok, higher_ok, moment_ok,
                     is_xfailing):
        try:
            distfn = getattr(stats, distname)
        except TypeError:
            distfn = distname
            distname = 'rv_histogram_instance'

        with npt.suppress_warnings() as sup:
            sup.filter(IntegrationWarning,
                       "The integral is probably divergent, or slowly convergent.")
            sup.filter(IntegrationWarning,
                       "The maximum number of subdivisions.")
            sup.filter(IntegrationWarning,
                       "The algorithm does not converge.")

            if is_xfailing:
                sup.filter(IntegrationWarning)

            m, v, s, k = distfn.stats(*arg, moments='mvsk')

            with np.errstate(all="ignore"):
                if normalization_ok:
                    check_normalization(distfn, arg, distname)

                if higher_ok:
                    check_mean_expect(distfn, arg, m, distname)
                    check_skew_expect(distfn, arg, m, v, s, distname)
                    check_var_expect(distfn, arg, m, v, distname)
                    check_kurt_expect(distfn, arg, m, v, k, distname)
>                   check_munp_expect(distfn, arg, distname)

arg        = (10,)
distfn     = <scipy.stats._continuous_distns.irwinhall_gen object at 0x00000207EB40AED0>
distname   = 'irwinhall'
higher_ok  = True
is_xfailing = False
k          = -0.12
m          = 5.0
moment_ok  = True
normalization_ok = True
s          = 0.0
sup        = <numpy.testing._private.utils.suppress_warnings object at 0x00000207ECED4B10>
v          = 0.8333333333333334

scipy\stats\tests\test_continuous_basic.py:348:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
scipy\stats\tests\common_tests.py:102: in check_munp_expect
    assert_allclose(res, ref, atol=1e-10, rtol=1e-4,
        args       = (10,)
        dist       = <scipy.stats._continuous_distns.irwinhall_gen object at 0x00000207EB40AED0>
        msg        = 'irwinhall'
        ref        = 4216.6666666727815
        res        = 5.183528709112784e+16
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<function assert_allclose.<locals>.compare at 0x00000207EC9239C0>, array(5.18352871e+16), array(4216.66666667))
kwds = {'equal_nan': True, 'err_msg': 'irwinhall - higher moment / _munp', 'header': 'Not equal to tolerance rtol=0.0001, atol=1e-10', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Not equal to tolerance rtol=0.0001, atol=1e-10
E           irwinhall - higher moment / _munp
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 5.18352871e+16
E           Max relative difference: 1.22929535e+13
E            x: array(5.183529e+16)
E            y: array(4216.666667)

args       = (<function assert_allclose.<locals>.compare at 0x00000207EC9239C0>, array(5.18352871e+16), array(4216.66666667))
func       = <function assert_array_compare at 0x000002076FAA7D80>
kwds       = {'equal_nan': True, 'err_msg': 'irwinhall - higher moment / _munp', 'header': 'Not equal to tolerance rtol=0.0001, atol=1e-10', 'verbose': True}
self       = <contextlib._GeneratorContextManager object at 0x000002076F9B1510>

..\miniforge3\envs\scipy-dev\Lib\contextlib.py:81: AssertionError
=============================================== short test summary info ===============================================
FAILED scipy/stats/tests/test_continuous_basic.py::test_moments[irwinhall-arg50-True-True-True-False] - AssertionError:
================================ 1 failed, 108 passed, 10 skipped, 2 xfailed in 26.85s ================================
```

When I remove this function override, the test passes because it falls back to the default `_munp` implementation. So the simplest fix is just to remove this unless you would like to work out the formula.

Reply:
Let's leave it out for now, and if you want to submit it as a follow-up PR, we could consider it.

Reply:
Well, ended up doing the derivation by hand and it turns out to be surprisingly simple in the end--let me see if I can add it in. Otherwise I'll just defer to the superclass.

@mdhaber we definitely don't do anything with cumulants, right? If not we probably should since they tend to work for cases where moments don't and converting between the two is tedious but doable.

Reply:
Feel free to make that suggestion in gh-15928. It is already on my mind to add support for distribution L-moments (you can see that I've drafted support for the distribution of order statistics), and I've had a PR open since the fall for a function to compute sample l-moments (gh-19475). Both of these are toward the goal of adding method of L-moments for fitting.

Reply:
> Well, ended up doing the derivation by hand and it turns out to be surprisingly simple in the end

You seemed interested in getting this PR merged sooner rather than later. Adding these will slow it down because it would require a maintainer not only to review the new code but also follow your derivations (which probably means deriving them oneself).